src/topology/controller.rs

```diff
@@ -93,6 +98,41 @@ impl TopologyController {
             }
         }
 
+        /*
+        Without this block
+        the first config loaded determines whether the api is
+        always on or always off
+        */
+        #[cfg(feature = "api")]
+        if new_config.api.enabled && self.api_server.is_none() {
+            use std::sync::atomic::AtomicBool;
+            debug!("Starting api server.");
+            self.api_server = match api::Server::start(
+                self.topology.config(),
+                self.topology.watch(),
+                Arc::<AtomicBool>::clone(&self.topology.running),
+                &Handle::current(),
+            ) {
+                Ok(api_server) => {
+                    emit!(ApiStarted {
+                        addr: new_config.api.address.unwrap(),
+                        playground: new_config.api.playground
+                    });
+
+                    Some(api_server)
+                }
+                Err(e) => {
+                    error!("An error occurred that Vector couldn't handle: {}.", e);
+                    return FatalError;
+                }
+            }
+        } else if !new_config.api.enabled {
```

Comment:
If this is reordered as the first condition, you can simplify the above to just `else if self.api_server.is_none()`.